pkg/client/api.go

```diff
@@ -81,7 +81,12 @@ func (c *httpAPIClient) Do(ctx context.Context, verb, endpoint string, query url
 	}()
 
 	if err != nil {
-		return APIResponse{}, err
+		return APIResponse{}, &Error{
+		  	Type: ErrBadResponse,
+		  	ErrorMsg:  err.Error(),
+			Query: query.Encode(),
```

Comment:
Can we decode this so it decodes the url encoding back to regular chars, e.g. `%20` back to `+`. It'll be more legible and easier to debug